### ISSUE TITLE
pgpainless-cli. Check pass phrase

### ISSUE THREAD

**Issue by DenBond7** [vanitasvitae/pgpainless#201]
For now, `./pgpainless-cli help` shows the following

```cmd
Usage: pgpainless-cli [COMMAND]
Commands:
  help                                 Displays help information about the
                                         specified command
  armor                                Add ASCII Armor to standard input
  dearmor                              Remove ASCII Armor from standard input
  decrypt                              Decrypt a message from standard input
  detach-inband-signature-and-message  Split a clearsigned message
  encrypt                              Encrypt a message from standard input
  extract-cert                         Extract a public key certificate from a
                                         secret key from standard input
  generate-key                         Generate a secret key
  sign                                 Create a detached signature on the data
                                         from standard input
  verify                               Verify a detached signature over the
                                         data from standard input
  version                              Display version information about the
                                         tool
```

It would be nice to have the ability to check a pass phrase for a provided private key.

`./pgpainless-cli check_passphrase 'file.asc' 'my pass phrase'`

```java
public static void main(String[] args) {
        System.out.println("args = " + Arrays.deepToString(args));
        File f = new File(args[0]);
        try {
            PGPSecretKeyRing secretKeyRing = PGPainless.readKeyRing().secretKeyRing(new FileInputStream(f));
            PGPainless.modifyKeyRing(secretKeyRing)
                    .changePassphraseFromOldPassphrase(Passphrase.fromPassword(args[1]))
                    .withSecureDefaultSettings()
                    .toNoPassphrase()
                    .done();
            System.out.println("Provided pass phrase: matched");
        } catch (Exception e) {
            System.out.println("Provided pass phrase: mismatch");
            System.out.println("Details:\n");
            e.printStackTrace();
        }
    }
```

**Comment by vanitasvitae** [comment]
While I agree that this feature would be useful, `pgpainless-cli` currently lacks support for passphrases completely.
This is because `pgpainless-cli` aims to implement the [Stateless OpenPGP Protocol](https://datatracker.ietf.org/doc/html/draft-dkg-openpgp-stateless-cli-02) which deliberately and explicitly states that

>    Secret key material should be in cleartext (that is, it should not be
   locked with a password).  If the secret key material is locked with a
   password, "sop" may fail with error "KEY_IS_PROTECTED".

It might be conceivable though to maybe add non-standard options, operations and flags in a separate binary at some point.

**Comment by DenBond7** [comment]
> It might be conceivable though to maybe add non-standard options, operations and flags in a separate binary at some point.

Agree. It can be non-standard options or even an external module and so on.

> Secret key material should be in cleartext (that is, it should not be
locked with a password)

Hmm... in that case maybe will be better to don't add such an option to keep `pgpainless-cli` clear as possible.

Anyway, it's not urgent. Just it would be nice to have such an option.

**Comment by vanitasvitae** [comment]
Closed in favor of #203
